### ISSUE TITLE
InvalidOperationException in integration test with LanguageClient using SemanticTokens

### ISSUE THREAD

**Issue by anthony-c-martin** [OmniSharp/csharp-language-server-protocol#516]
I'm trying to add an integration test for semantic tokens to our codebase, but I'm hitting a cryptic exception on the semantic tokens request I make from client -> server:

```
Exception thrown: 'System.InvalidOperationException' in System.Private.CoreLib.dll: 'Sequence contains no elements.'
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Bicep.LangServer.IntegrationTests.SemanticTokenTests.<Duplicate_tokens_are_not_returned>d__4.MoveNext() in /Users/ant/Code/bicep/src/Bicep.LangServer.IntegrationTests/SemanticTokenTests.cs:line 32
```

This is the PR I'm trying to introduce this test with: https://github.com/Azure/bicep/pull/1485/files - the test is under [SemanticTokenTests.cs](https://github.com/Azure/bicep/blob/f1df698ba1504a12f8297de93b3a47627d1aa4d0/src/Bicep.LangServer.IntegrationTests/SemanticTokenTests.cs#L29-L36).

I'm not doing anything fancy to initialize the LanguageClient (init code [here](https://github.com/Azure/bicep/blob/3c12faa1d2ec14bbee55cdb325ab46eeb67750b0/src/Bicep.LangServer.IntegrationTests/Helpers/IntegrationTestHelper.cs#L90)), and we use this same mechanism to test out various other language server features. I'm wondering if there's perhaps some mechanism to register capabilities that I'm missing?

Any pointers would be very helpful!

**Comment by david-driscoll** [comment]
@anthony-c-martin sorry I didn't see this before.  I'll take a look, sounds like a `Single` call

**Comment by anthony-c-martin** [comment]
> @anthony-c-martin sorry I didn't see this before. I'll take a look, sounds like a `Single` call

I've spent a while debugging this, but I'm unfortunately not at all familiar with the System.Reactive library being used. I couldn't find anything indicating the exception was occurring in our code, as I was able to step through the handler logic with "break on all exceptions" - and this is the only exception thrown.

Unfortunately the above stack trace doesn't give me any clues - it seems like all the originating call site information is being lost (I believe somewhat related to https://github.com/dotnet/reactive/issues/1235).
